react/features/authentication/middleware.ts

```diff
@@ -100,26 +106,72 @@ MiddlewareRegistry.register(store => next => action => {
         if (_isWaitingForOwner(store)) {
             store.dispatch(stopWaitForOwner());
         }
-        _hideLoginDialog(store);
+        store.dispatch(hideLoginDialog());
         break;
 
     case CONFERENCE_LEFT:
         store.dispatch(stopWaitForOwner());
         break;
 
     case CONNECTION_ESTABLISHED:
-        _hideLoginDialog(store);
+        store.dispatch(hideLoginDialog());
         break;
 
     case CONNECTION_FAILED: {
         const { error } = action;
+        const state = store.getState();
+        const { jwt } = state['features/base/jwt'];
 
         if (error
                 && error.name === JitsiConnectionErrors.PASSWORD_REQUIRED
-                && typeof error.recoverable === 'undefined') {
+                && typeof error.recoverable === 'undefined'
+                && !jwt) {
             error.recoverable = true;
+
+            const config = state['features/base/config'];
+            const room = getBackendSafeRoomName(state['features/base/conference'].room);
+
+            // this is only for web
+            if (isTokenAuthEnabled(config) && typeof APP !== 'undefined') {
+                // FIXME: This method will not preserve the other URL params that were originally passed.
+                // redirectToTokenAuthService
+                window.location.href = getTokenAuthUrl(config)(room, false);
+            } else {
+                store.dispatch(openLoginDialog());
+            }
+        }
+
+        break;
+    }
+
+    case LOGIN: {
+        const state = store.getState();
+        const config = state['features/base/config'];
+        const room = getBackendSafeRoomName(state['features/base/conference'].room);
+
+        // this is only for web
+        if (isTokenAuthEnabled(config) && typeof APP !== 'undefined') {
+            // FIXME: This method will not preserve the other URL params that were originally passed.
+            // redirectToTokenAuthService
+            window.location.href = getTokenAuthUrl(config)(room, false);
+        } else {
             store.dispatch(openLoginDialog());
```

Comment:
Ditto.

Reply:
Here, this is when you click the login button in the profile tab. We show the login window asking for user and pass or we just redirect to the token page to get the jwt.